### PR TITLE
Add Ragtag tool

### PR DESCRIPTION
**Pull request by gallardoalba**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/ragtag/macros.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            <requirement type="package" version="@TOOL_VERSION@">ragtag</requirement>
+        </requirements>
+    </xml>
+    <xml name="citations">
+        <citations>
+            <citation type="doi">10.1186/s13059-019-1829-6</citation>
+        </citations>
+    </xml>
+    <xml name="input_options">
+        <param name="reference" type="data" format="fasta,fasta.bgzip" label="Reference FASTA file"/>
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply tools/ragtag/macros.xml:20]: It is what the documentation says. I'll check if FASTA.gz is also allowed.

File: tools/ragtag/ragtag.xml [hunk 2]
```diff
@@ -0,0 +1,543 @@
...
+        --gap-func $mode_conditional.merging_options.function_merging
+        #if $mode_conditional.hic_options.b
+            -b $mode_conditional.hic_options.b
+            -r $mode_conditional.hic_options.r
+            -p $mode_conditional.hic_options.p
+        #end if 
+    #end if
+    -o ./
+    #if $mode_conditional.mode_option != 'merge'
+        -t \${GALAXY_SLOTS:-2}
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply tools/ragtag/ragtag.xml:93]: Yes, I'll check the last Rastag version if it has changed.

[the remark this answers is not in the document]
Comment (bgruening) [reply tools/ragtag/ragtag.xml:93]: Ok, this seems to be ok!

File: tools/ragtag/ragtag.xml [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+                        </when>
+                        <when value="false"/>
+                    </conditional>
+                    <conditional name="unplaced_conditional">
+                        <param name="unplaced_option" type="select" label="Concatenate unplaced contigs and make 'chr0' (-C)">
+                            <option value="true">Enabled</option>
+                            <option value="false" selected="true">Disabled</option>
+                        </param>
+                        <when value="true">
+                            <param argument="-J" type="data" format="txt" optional="true" label="List of query headers to leave unplaceds and exclude form 'chr0'"/>
```
[the remark this answers is not in the document]
Comment (gallardoalba) [reply tools/ragtag/ragtag.xml:188]: No, I don't think so, because each header should be in a different line.

**Comment by gallardoalba** [comment]
When using non-optional checklists, it triggers this error:
`.. ERROR: Select [output_scaffold] display="checkboxes" is incompatible with  optional="false", remove the display attribute`

**Comment by bgruening** [comment]
> ERROR: Select [output_scaffold] display="checkboxes" is incompatible with optional="false", remove the display attribute

Can you not use a plain select?

**Comment by gallardoalba** [comment]
> > ERROR: Select [output_scaffold] display="checkboxes" is incompatible with optional="false", remove the display attribute
> 
> Can you not use a plain select?

Sure.

**Review by bgruening — APPROVED** [review]

**Pull request merged**